.github/workflows/rust.yml

```diff
@@ -1,121 +1,88 @@
-name:                  Test Suite
+name:                              Compilation and Testing Suite
 
 on:
   pull_request:
   push:
     branches:
       - master
-      - stable
     tags:
       - v*
     paths-ignore:
       - 'README.md'
+      - diagrams/
+  schedule:                        #Nightly build
```

Comment:
I think it would be better to have weekly builds instead (even tagged and published on Github), we don't move fast enough for daily builds.